internal/daemon/daemon.go

```diff
@@ -111,11 +134,9 @@ func NewDaemon(project string, version string) *Daemon {
 	return d
 }
 
-// Run initializes the Daemon with the given configuration, starts the database, and blocks until the daemon is cancelled.
-// - `extensionsSchema` is a list of schema updates in the order that they should be applied.
-// - `extensionServers` is a list of rest.Server that will be initialized and managed by microcluster.
-// - `hooks` are a set of functions that trigger at certain points during cluster communication.
-func (d *Daemon) Run(ctx context.Context, listenAddress string, heartbeatInterval time.Duration, stateDir string, socketGroup string, extensionsSchema []schema.Update, apiExtensions []string, extensionServers map[string]rest.Server, hooks *state.Hooks) error {
+// Run initializes the Daemon with the given configuration, starts the database,
+// and blocks until the daemon is cancelled.
+func (d *Daemon) Run(ctx context.Context, stateDir string, socketGroup string, args Args) error {
```

Comment:
How come `SocketGroup` isn't a daemon arg?

Reply:
Because we use it to initialize the default OS [here](https://github.com/canonical/microcluster/blob/main/microcluster/app.go#L68). Looking a little closer, I'm not convinced that we should be exposing `SocketGroup` via the OS struct at all; it's not relevant to the permissions of anything other than the socket.

I agree; my vote is we remove `SocketGroup` from `sys.OS` and include it in `DaemonArgs`. For now I'll pass it straight through to `startUnixServer`.